src/app/vertextool/qgsvertexeditor.cpp

```diff
@@ -300,8 +294,26 @@ QgsVertexEditor::QgsVertexEditor(
 
   setWidget( mTableView );
 
-  connect( mSelectedFeature, &QgsSelectedFeature::selectionChanged, this, &QgsVertexEditor::updateTableSelection );
   connect( mTableView->selectionModel(), &QItemSelectionModel::selectionChanged, this, &QgsVertexEditor::updateVertexSelection );
+
+  updateEditor( layer, selectedFeature );
+}
+
+void QgsVertexEditor::updateEditor( QgsVectorLayer *layer, QgsSelectedFeature *selectedFeature )
+{
+  if ( mVertexModel )
```

Comment:
Candidate for `std::unique_ptr`?

Reply:
Yeah I can change it if that is better here.

Reply:
@m-kuhn I looked into unique_ptr here - it's actually not applicable, because this model *has* to be parented to the widget, and if we use a unique_ptr we'll run the risk of the unique_ptr getting deleted after the model is deleted by Qt. (seen elsewhere when changing parented QObjects to unique_ptr)